### PR TITLE
feat(wallet): Add Info Tooltip to Price Impact

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 

- Adds a `Info Tooltip` description to the `Price Impact` section for `Swap` and `Bridge`.

Resolves <https://github.com/brave/brave-browser/issues/39683>

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Open the `Wallet` and navigate to the `Swap` or `Bridge` screen
2. Select some tokens and get a quote
3. Expand the `Advanced details` in the `Quote Info`
4. Hover over the `Tooltip` for the `Price Impact` section to view the description info.

Screenshot 20

Screenshot 21

### CODE REVIEW

**Review by josheleonard — APPROVED** [review]
++

**Review by yrliou — APPROVED** [review]
core++

File: components/resources/wallet_strings.grdp [hunk 1]
```diff
@@ -125,6 +125,7 @@
   <message name="IDS_BRAVE_WALLET_ROUTE" desc="Swap route label">Route</message>
   <message name="IDS_BRAVE_WALLET_MAX_SLIPPAGE" desc="Swap max slippage label">Max slippage</message>
   <message name="IDS_BRAVE_WALLET_SLIPPAGE_DESCRIPTION" desc="Swap max slippage description">When the deviation between the price of the transaction you submitted and the price at the time of the transaction on chain is greater than this set value, the transaction will fail.</message>
+  <message name="IDS_BRAVE_WALLET_PRICE_IMPACT_DESCRIPTION" desc="Swap price impact description">Price impact refers to the difference between the expected execution price based on market price and trade size.</message>
```
Comment (mkarolin) [inline components/resources/wallet_strings.grdp:128]: This seems a bit confusing. The difference between what and what?

Comment (Douglashdaniel) [reply components/resources/wallet_strings.grdp]: @mkarolin I have updated the copy for this! https://github.com/brave/brave-core/compare/4e59ebe958974fc5e99a294dd298fdf05e7f54aa..3faf38adac489c16d52914a31c10aa843c576e9e

**Review by mkarolin — APPROVED** [review]
`strings`++

**Pull request merged**